File: Code/server/app/post/comment.py
from .funcs import is_exist
from typing import List
from app.models import engine, Comment, ObjectId, User, CommentFunc
from fastapi import APIRouter, Depends
from app.config.depends import get_current_user

router = APIRouter(prefix="/comment")

@router.post('/create/', response_model=Comment)
async def create(postId: ObjectId, content: str, current_user: User = Depends(get_current_user)):
    post = await is_exist(postId)
    new_comment = Comment(
        content=content,
        commented=post,
        commenter=current_user,
    )
    post.comments = post.comments + 1
    await engine.save_all([new_comment, post])
    return new_comment

@router.get('/list/')
async def list(postId: ObjectId, page: int = 0, current_user: User = Depends(get_current_user)):
    comments = await engine.find(Comment, Comment.commented == postId, skip=(page * 13), limit=13, sort=Comment.created_at.desc())
    list_comments = [await CommentFunc.get_classic_data(x) for x in comments]
    return list_comments

File: Code/petiapp/lib/models/comment.dart
import 'package:petiapp/models/Post.dart';
import 'package:petiapp/models/User.dart';

class Comment {
  Comment({
    required this.content,
    this.commented,
    required this.commenter,
    required this.createdAt,
    this.id,
  });

  String content;
  Post? commented;
  User commenter;
  DateTime createdAt;
  String? id;

  factory Comment.fromMap(Map<String, dynamic> json) => Comment(
        content: json["content"],
        commented:
            json["commented"] == null ? null : Post.fromMap(json["commented"]),
        commenter: User.fromMap(json["commenter"]),
        createdAt: DateTime.parse(json["created_at"]),
        id: json["id"],
      );
}
